src/tls.c

```diff
@@ -677,12 +678,69 @@ static void updateSSLState(connection *conn_) {
     updatePendingData(conn);
 }
 
+static int getCertFieldByName(X509 *cert, const char *field, char *out, size_t outlen) {
+    if (!cert || !field || !out) return 0;
+
+    int nid = -1;
+
+    if (!strcasecmp(field, "CN"))
+        nid = NID_commonName;
+    else if (!strcasecmp(field, "O"))
+        nid = NID_organizationName;
+    // Add more mappings here as needed
+
+    if (nid == -1) return 0;
+
+    X509_NAME *subject = X509_get_subject_name(cert);
+    if (!subject) return 0;
+
+    return X509_NAME_get_text_by_NID(subject, nid, out, outlen) > 0;
+}
+
+static void autoAuthenticateClientFromCert(tls_connection *conn) {
+    if (!conn || !SSL_is_init_finished(conn->ssl)) return;
+
+    const char *field = server.tls_ctx_config.client_auth_field;
+    if (!field) return;
+
+    serverLog(LL_NOTICE, "TLS: Config Auto Auth Field = %s", field);
+
+    X509 *cert = SSL_get_peer_certificate(conn->ssl);
+    if (!cert) {
+        serverLog(LL_WARNING, "TLS: No peer certificate found after handshake");
```

Comment:
WARNING is a serious log level. It means the server is in trouble and needs an administrator's help.

Missing cert is not an error. Client certificate can be configured to be optional. I think we shouldn't log anything here, not even a notice.

Reply:
Done updated